src/benchmarks/real-world/ILLink/MSBuildBenchmark.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.IO;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Engines;
+
+namespace ILLinkBenchmarks;
+
+[BenchmarkCategory("ILLink")]
+[SimpleJob(RunStrategy.Monitoring, launchCount: 1, warmupCount: 3, targetCount: 20)]
+public class RunMSBuildPublish
+{
+    string _projectFilePath;
+    string _projectOutputPath;
+    string _iterationPublishPath;
+    string _linkSemaphore;
+
+    [GlobalSetup(Targets = new[] { nameof(LinkHelloWorld) })]
+    public void LinkHelloWorldGlobalSetup()
+    {
+        _projectFilePath = Environment.GetEnvironmentVariable("ILLINK_SAMPLE_PROJECT");
+        _projectOutputPath = Utilities.PublishSampleProject(_projectFilePath);
+        _linkSemaphore = Path.Combine(_projectOutputPath, "link.semaphore");
+        return;
+    }
+
+    [GlobalCleanup(Targets=new[] {nameof(LinkHelloWorld) })]
+    public void LinkHelloWorldGlobalCleanup()
+    {
+        Directory.Delete(_projectOutputPath, recursive: true);
+    }
+
+    [Benchmark]
+    [BenchmarkCategory("ILLink")]
+    public string LinkHelloWorld()
+    {
+        _iterationPublishPath = Utilities.PublishSampleProject(_projectFilePath, "--no-build", "/p:PublishTrimmed=true", $"/p:_LinkSemaphore=\"{_linkSemaphore}\"");
+        return _iterationPublishPath;
+    }
+
+    [IterationCleanup(Targets = new[] { nameof(LinkHelloWorld) })]
+    public void LinkHelloWorldIterationCleanup ()
+    {
+        Directory.Delete(_iterationPublishPath, recursive: true);
+        File.Delete(_linkSemaphore);
```

Comment:
If you delete the `_linkSemaphore` here subsequent iterations will not have the file, since the `GlobalSetup` method will not be run again. You might want to move the sempahore creation into an IterationSetup method.

Reply:
The `_linkSemaphore` is used as the signal for incremental builds to avoid running the linker, so it needs to be deleted after every iteration.